### ISSUE TITLE
[BUG] Use default AWS credentials chain

### ISSUE THREAD

**Issue by caiorcferreira** [benthosdev/benthos#1983]
Currently, benthos creates it's AWS Session starting from a `aws.NewConfig()` call, which provides a pointer to an empty config.

However, in edge cases like when using [KIAM](https://github.com/uswitch/kiam), this does not work. Even using `from_ec2_role` isn't enough, as it seams AWS SDK looks for instance information in a predefined path that does not exist in the context of a Kubernetes container.

Every application developed using the AWS SDK and `github.com/aws/aws-sdk-go-v2/config.LoadDefaultConfig` has worked fine with KIAM and with this benthos would hold the AWS contract for credential configuration.

Finally, the same config customizations done today would still be applicable with this new configuration loading.

I can open a PR you think this sounds like a good path

Thanks

**Comment by Jeffail** [comment]
Hey @caiorcferreira, I'm not sure I fully understand the solution, would we need to migrate to the v2 AWS API? As long as it's backwards compatible them I'm happy to accept a fix.

**Comment by caiorcferreira** [comment]
After another of troubleshooting I was able to figure out what was happing. For future reference, what happened was:

Because of the error message `Failed to read message: AccessDenied: Access Denied\n\tstatus code: 403, request id: REAT4SFJ99ZDDBT8, host id: abdcde" @service=benthos label=input path=root.input` I though that the problem was consuming messages from SQS. Turns out I was wrong, as `Failed to read message` is a static message added by benthos [here](https://github.com/benthosdev/benthos/blob/c93d4ff119b4cf870e8a6ae383db5584263226cd/internal/component/input/async_reader.go#L172).

Benthos was failing because the objects in the input bucket didn't have the correct ACL and were still owned by the writer.

It would be nice to wrap error with the action being taken to improve clarity. Maybe later I open a PR with this.

Thanks
